service/worker/failovermanager/workflow_test.go

```diff
@@ -194,28 +210,46 @@ func (s *failoverWorkflowTestSuite) TestWorkflow_Pause() {
 		BatchFailoverSize: 2,
 		Domains:           domains,
 	}
-	env.ExecuteWorkflow(WorkflowTypeName, params)
+	s.workflowEnv.ExecuteWorkflow(WorkflowTypeName, params)
 
 	var result FailoverResult
-	s.NoError(env.GetWorkflowResult(&result))
+	s.NoError(s.workflowEnv.GetWorkflowResult(&result))
 	s.Equal(mockFailoverActivityResult.SuccessDomains, result.SuccessDomains)
 }
 
-func (s *failoverWorkflowTestSuite) assertQueryState(env *testsuite.TestWorkflowEnvironment, expectedState string) {
-	queryResult, err := env.QueryWorkflow(QueryType)
+func (s *failoverWorkflowTestSuite) TestWorkflow_WithDrillWaitTime_Success() {
+	domains := []string{"d1"}
+	mockFailoverActivityResult := &FailoverActivityResult{
+		SuccessDomains: []string{"d1"},
+	}
+	s.workflowEnv.OnActivity(getDomainsActivityName, mock.Anything, mock.Anything).Return(domains, nil)
+	s.workflowEnv.OnActivity(failoverActivityName, mock.Anything, mock.Anything).Return(mockFailoverActivityResult, nil)
+	params := &FailoverParams{
+		TargetCluster: "t",
+		SourceCluster: "s",
+		DrillWaitTime: 1,
```

Comment:
nit: probably use `SetOnTimerFiredListener` to validate that the wait timer is scheduled and fired.

Reply:
Added